test/XRP/issued-currency-client-integration.test.ts

```diff
@@ -420,4 +420,34 @@ describe('IssuedCurrencyClient Integration Tests', function (): void {
     assert.equal(createdTrustLine.limit, trustLineLimit)
     assert.equal(createdTrustLine.currency, trustLineCurrency)
   })
+
+  it('authorizeTrustline - valid account', async function (): Promise<void> {
+    this.timeout(timeoutMs)
+    const accountToTrust = await XRPTestUtils.randomWalletFromFaucet()
+
+    // GIVEN an existing testnet account requiring authorized trust lines
+    // and another account
+    await issuedCurrencyClient.requireAuthorizedTrustlines(wallet)
+
+    const trustLineCurrency = 'USD'
+    // WHEN a trust line is authorized with another account
+    await issuedCurrencyClient.authorizeTrustLine(
+      accountToTrust.getAddress(),
+      'USD',
+      wallet,
+    )
+
+    const trustLines = await issuedCurrencyClient.getTrustLines(
+      wallet.getAddress(),
+    )
+
+    const [createdTrustLine] = trustLines
+    const classicAddress = XrpUtils.decodeXAddress(accountToTrust.getAddress())!
+
+    // THEN an authorized trust line is created between the wallet and the other account.
+    assert.equal(createdTrustLine.account, classicAddress.address)
+    assert.equal(createdTrustLine.limit, '0')
+    assert.equal(createdTrustLine.currency, trustLineCurrency)
+    assert.isTrue(createdTrustLine.authorized)
+  })
```

Comment:
Should we add any additional test cases here?  For example, what if we created a test where we first establish a non-zero trustline from the customer to the issuer, and THEN have the issuer call `authorizeTrustLine` and observe that the trust line exists and has a non-0 limit?  Or a test where the issuer requires auth trust lines, does not authorize a line to a customer, and then tries to send money to the customer and it fails (<-- actually, this case I think belongs more in `sendIssuedCurrency` and is currently an open TODO for me, pending this PR, so I can take care of that one).

Reply:
yeah I've been having a bit of hard time trying to think about where the boundaries lie in terms of logic that should be tested here v.s. logic that is part of the XRPL and shouldn't be tested here.... I'm fine leaving this test file as-is for now.  A demo will be the real test.